File: Benchmarking/noscaling/euler3/euler3.java
import java.util.*;

public class euler3
{
	public static void main (String[] args)
	{
		prime_factors(Long.parseLong(args[0]));
	}
	
	private static List<Long> prime_factors(long n)
	{
		List<Long> factors = new ArrayList<>();
		long d = 2;
		while(n > 1)
		{
			while(n % d == 0)
			{
				factors.add(d);
				n /= d;
			}
			d += 1;
		}
		return factors;
	}
}

File: Benchmarking/noscaling/euler3/euler3.cpp
//
//  main.cpp
//  Euler3
//
//  Created by Sonni on 19/05/16.
//  Copyright © 2016 Sonni Jensen. All rights reserved.
//

#include <iostream>
#include <vector>

void printVector(std::vector<unsigned int>& v)
{
    for (unsigned int i = 0; i < v.size(); i++)
        std::cout << v[i] << ", ";
}

void prime_factors(unsigned long n)
{
    std::vector<unsigned int> factors;
    unsigned int d = 2;
    while (n > 1)
    {
        while (n % d == 0)
        {
            factors.push_back(d);
            n /= d;
        }
        d += 1;
    }
    printVector(factors);
}

int main(int argc, const char * argv[]) {
    prime_factors(atoi(argv[1]));
    return 0;
}
